clients/node/client-rds-data-node/protocol/AwsRestJson1_1.ts

```diff
@@ -83,49 +88,36 @@ export function executeStatementAwsRestJson1_1Deserialize(
 function executeStatementAwsRestJson1_1DeserializeError(
   output: HttpResponse
 ): Promise<ExecuteStatementResponse> {
-  let data = parseBody(output);
-  if (output.statusCode === 400 && data.dbConnectionId !== undefined) {
-    return Promise.reject({
-      __type: "com.amazon.rdsdataservice#StatementTimeoutException",
-      $name: "StatementTimeoutException",
-      $fault: "client",
-      message: data.message,
-      dbConnectionId: data.dbConnectionId
-    });
-  }
-
-  if (output.statusCode === 400) {
-    return Promise.reject({
-      __type: "com.amazon.rdsdataservice#BadRequestException",
-      $name: "BadRequestException",
-      $fault: "client",
-      message: data.message
-    });
-  }
-
-  if (output.statusCode === 403) {
-    return Promise.reject({
-      __type: "com.amazon.rdsdataservice#ForbiddenException",
-      $name: "ForbiddenException",
-      $fault: "client",
-      message: data.message
-    });
-  }
-
-  if (output.statusCode === 500) {
-    return Promise.reject({
-      __type: "com.amazon.rdsdataservice#InternalServerErrorException",
-      $name: "InternalServerErrorException",
-      $fault: "server"
-    });
-  }
+  const data = parseBody(output);
+
+  const errorMap: { [index: string]: any } = {
```

Comment:
This could probably be a switch rather than a map to not need to recreate the value (I think that's what this does anyway). Then in your switch, you can use fall-through to catch both the fully-qualified and relative versions.

Reply:
swapped to switch in ec7ce7df0eaa70f6e77ab41f8a3f25d0507ebd95